packages/joy-types/src/hiring/index.ts

```diff
@@ -0,0 +1,278 @@
+
+export class ActivateOpeningAt extends Enum {
+    constructor (value?: any, index?: number) {
+        super(
+          {
+            CurrentBlock,
+            ExactBlock,
+          },
+          value, index);
+      }
+}
+
+export class ApplicationDeactivationCause extends Enum {
+    constructor (value?: any, index?: number) {
+        super(
+          [
+            'External',
+            'Hired',
+            'NotHired',
+            'CrowdedOut',
+            'OpeningCancelled',
+            'ReviewPeriodExpired',
+            'OpeningFilled',
+          ],
+          value, index);
+    }
+};
+
+export type UnstakingApplicationStageType = {
+    deactivation_initiated: BlockNumber,
+    case: ApplicationDeactivationCause
+};
+export class UnstakingApplicationStage extends JoyStruct<UnstakingApplicationStageType> {
+    constructor (value?: UnstakingApplicationStageType) {
+      super({
+        deactivation_initiated: u32, // BlockNumber
+        cause: ApplicationDeactivationCause,
+      }, value);
+    }
+};
+
+export type InactiveApplicationStageType = {
+    deactivation_initiated: BlockNumber,
+    deactivated: BlockNumber,
+    case: ApplicationDeactivationCause
+};
+export class InactiveApplicationStage extends JoyStruct<InactiveApplicationStageType> {
+    constructor (value?: InactiveApplicationStageType) {
+      super({
+        deactivation_initiated: u32, // BlockNumber
+        deactivated: u32,
+        cause: ApplicationDeactivationCause,
+      }, value);
+    }
+};
+
+export class ActiveApplicationStage extends Null {};
+
+export class ApplicationStage extends Enum {
+    constructor (value?: any, index?: number) {
+      super(
+        {
+          'Active': Null,
+          'Unstaking': UnstakingApplicationStage,
+          'Inactive': InactiveApplicationStage,
+        },
+        value, index);
+    }
+}
+
+export type IApplicationRationingPolicy = {
+    max_active_applicants: u32,
+};
+export class ApplicationRationingPolicy extends JoyStruct<IApplicationRationingPolicy> {
+    constructor (value?: IApplicationRationingPolicy) {
+        super({
+            max_active_applicants: u32,
+        }, value);
+    }
+};
+
+export type WaitingToBeingOpeningStageVariantType = {
+    begins_at_block: BlockNumber,
+};
+export class WaitingToBeingOpeningStageVariant extends JoyStruct<WaitingToBeingOpeningStageVariantType> {
+    constructor (value?: WaitingToBeingOpeningStageVariantType) {
+        super({
+            begins_at_block: u32,
+        }, value);
+    }
+};
+
+export class OpeningDeactivationCause extends Enum {
+    constructor (value?: any, index?: number) {
+        super(
+          [
+            'CancelledBeforeActivation',
+            'CancelledAcceptingApplications',
+            'CancelledInReviewPeriod',
+            'ReviewPeriodExpired',
+            'Filled',
+          ],
+          value, index);
+    }
+};
+
+export type IAcceptingApplications = {
```

Comment:
```suggestion
export type AcceptingApplicationsType = {
```

I already used a convention with appending `Types` in other types.
So to be consistent it makes sense to follow that convention.
But it's up to you actually. Just saying :) 

To me, this `I` prefix looks so Java-ish.

Reply:
Yes I saw this. Honestly I found adding I at the start made it clearer especially for some of the type names that were enum variants, it just became confusing. Also I don't think its even necessary for this type to be exported (is that correct), which perhaps makes the choice not so significant?

Reply:
This actually brings up an important point. I noticed in polkadot types most of the runtime types are defined using typescript `interface` where as we are using `type`. Is there a particular reason we aren't defining our types in the same way?